store_test.go

```diff
@@ -485,3 +486,107 @@ func TestWithSplitStore(t *testing.T) {
 	store.Free()
 	store.Free()
 }
+
+func TestStoreList(t *testing.T) {
+	reexec.Init()
+	wd := t.TempDir()
+
+	pullOpts := map[string]string{"Test1": "test1", "Test2": "test2"}
+	store, err := GetStore(StoreOptions{
+		RunRoot:            filepath.Join(wd, "run"),
+		GraphRoot:          filepath.Join(wd, "root"),
+		GraphDriverName:    "vfs",
+		GraphDriverOptions: []string{},
+		UIDMap: []idtools.IDMap{{
+			ContainerID: 0,
+			HostID:      os.Getuid(),
+			Size:        1,
+		}},
+		GIDMap: []idtools.IDMap{{
+			ContainerID: 0,
+			HostID:      os.Getgid(),
+			Size:        1,
+		}},
+		PullOptions: pullOpts,
+	})
+	require.NoError(t, err)
+
+	root := store.RunRoot()
+	require.NotNil(t, root)
+
+	_, err = store.GraphDriver()
+	require.Nil(t, err)
+
+	_, err = store.CreateLayer("Layer", "", nil, "", false, nil)
+	require.NoError(t, err)
+
+	_, err = store.CreateImage("Image", nil, "Layer", "", nil)
+	require.NoError(t, err)
+
+	_, err = store.CreateContainer("Container", nil, "Image", "", "", nil)
+	require.NoError(t, err)
+
+	tests := []struct {
+		options         ListOptions
+		layerCounts     int
+		imageCounts     int
+		containerCounts int
+	}{
+		{
+			options: ListOptions{
+				Layers:     true,
+				Images:     true,
+				Containers: true,
+			},
+			layerCounts:     3,
+			imageCounts:     1,
+			containerCounts: 1,
+		},
+
+		{
+			options: ListOptions{
+				Layers:     true,
+				Images:     false,
+				Containers: false,
+			},
+			layerCounts:     3,
+			imageCounts:     0,
+			containerCounts: 0,
+		},
+
+		{
+			options: ListOptions{
+				Layers:     false,
+				Images:     true,
+				Containers: false,
+			},
+			layerCounts:     0,
+			imageCounts:     1,
+			containerCounts: 0,
+		},
+
+		{
+			options: ListOptions{
+				Layers:     false,
+				Images:     false,
+				Containers: true,
+			},
+			layerCounts:     0,
+			imageCounts:     0,
+			containerCounts: 1,
+		},
+	}
+
+	for _, test := range tests {
+		listResults, err := store.List(test.options)
+		require.NoError(t, err)
+		require.Equal(t, test.layerCounts, len(listResults.Layers))
+		require.Equal(t, test.imageCounts, len(listResults.Images))
+		require.Equal(t, test.containerCounts, len(listResults.Containers))
```

Comment:
Non-blocking, this works fine:
- `require` for things that should fail the entire execution; `assert` for failures that allow continuing.
- `assert.Len()` exists